src/main/java/de/themoep/snap/forwarding/listener/ChatListener.java

```diff
@@ -33,10 +33,16 @@ public ChatListener(Snap snap) {
     @Subscribe
     public void on(PlayerChatEvent event) {
         SnapPlayer player = snap.getPlayer(event.getPlayer());
-        ChatEvent e = snap.getBungeeAdapter().getPluginManager().callEvent(new ChatEvent(player, player.getServer(), event.getMessage()));
+
+        ChatEvent e = new ChatEvent(
+                player, player.getServer(),
+                event.getResult().getMessage().orElse(event.getMessage())
+        );
+        e.setCancelled(!event.getResult().isAllowed());
+        snap.getBungeeAdapter().getPluginManager().callEvent(e);
         if (e.isCancelled()) {
             event.setResult(PlayerChatEvent.ChatResult.denied());
-        } else if (!e.getMessage().equals(event.getMessage())) {
```

Comment:
What's the reason behind removing that check?

Reply:
If the PlayerChatEvent is disallowed, and then a BungeeCord plugin undoes the cancelllation ("uncancels" the event) Snap needs to re-allow the PlayerChatEvent. In other words, Snap must transmit the exact cancellation status from BungeeCord to Velocity.

Reply:
I was referring to the line I commented on namely you updating the message all the time even when it didn't change.

Reply:
Actually now that I think about it this is necessary to un-cancel the event so yeah, checking that wouldn't make sense.